### PR TITLE
Update test.html to be more compliant/accessible

### PR DESCRIPTION
**Pull request by martynhoyer**
Just something I adjusted while testing which I thought the project could benefit from.

All inputs now have an `id` (since that's what `for` should be looking at, even though sometimes it works with `name`), and checkbox/radio groups are now improved with proper `<fieldset>`/`<legend>` combos and `<label>`s to expand the hit area for them.

### CODE REVIEW

**Comment by martynhoyer** [comment]
Just to note, it looks like 5 commits but it's actually just the one: 9cd1779. I just did it before the previous PR got merged, so it's got a bit messy when I merged upstream back.

File: test.html [hunk 1]
```diff
@@ -56,45 +56,45 @@
...
     <!--radio group-->
     <fieldset class="pure-group">
-      <label for="radio-group">OK, Now Really Choose:</label>
-      <input type="radio" name="radio-group" value="js" />Javascript
-      <input type="radio" name="radio-group" value="java" />Java
-      <input type="radio" name="radio-group" value="c++" />C++
-      <input type="radio" name="radio-group" value="python" checked />Python
-      <input type="radio" name="radio-group" value="other" />Other
+      <legend>OK, Now Really Choose:</legend>
+      <input id="radio-group--js" type="radio" name="radio-group" value="js" /> <label for="radio-group--javascript">Javascript</label>
```
Comment (mckennapsean) [inline test.html]: Just a minor bug in the `id` vs what's in the `for` here, everything else seems to work fine with the labels!

**Comment by martynhoyer** [comment]
> Quick question, is there a reason you need ID set on all form elements? From what I understand, it is really only needed when you have multiple labels on a single form element, and the extra code may not be needed for a single input box/option. Just a thought, but let me know what you think (the Javascript file checks for the name, not ID).

The `<label>`'s `for` attribute should be associated with the ID, so it should generally always be there in my opinion. I feel like it's good to get into the habit of adding it all the time even if it's not technically required (like when nesting the `<input>` as a child element of the `<label>`)

File: test.html (hunk 1 above)
Comment (martynhoyer) [inline test.html]: Good spot, poor consistency on my part! I'll fix that!

**Comment by mckennapsean** [comment]
Cool, thanks for the PR to improve the test file!

**Pull request merged**